hail/src/main/scala/is/hail/io/BufferSpecs.scala

```diff
@@ -151,6 +153,23 @@ final case class LZ4FastBlockBufferSpec(blockSize: Int, child: BlockBufferSpec)
   def typeName = "LZ4FastBlockBufferSpec"
 }
 
+final case class LZ4SizeBasedFastBlockBufferSpec(blockSize: Int, minCompressionSize: Int, child: BlockBufferSpec)
```

Comment:
Instead of making this LZ4SizeBasedFast..., can you put the hc vs fast in a parameter to the BufferSpec?  That will make it more flexible.